src/ProductDataProvider.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LupaSearch\LupaSearchPlugin;
+
+use PrestaShop\PrestaShop\Adapter\Entity\Db;
+use Context;
+use Product;
+
+class ProductDataProvider
+{
+    private $queryProvider;
+
+    public function __construct(QueryProvider $queryProvider = null)
+    {
+        $this->queryProvider = $queryProvider ?? new QueryProvider();
+    }
+
+    public function getFormattedProducts(int $page = 1, int $limit = 20): array
```

Comment:
Needs a lot of exception handling. For example `$productId = $product['id_product'];` if there is no 'id_product' you'll get Undefined array key "id_product".